### PR TITLE
Remove references to last selected points if removed with overlap.

### LINKED ISSUE #4380: Remove Overlap ASAN violation/crash
Importing this SVG file into a char: [blah3.svg.txt](https://github.com/fontforge/fontforge/files/4767592/blah3.svg.txt), followed by Control-A to select all points, followed by running Remove Overlap results in this ASAN address violation:

```
=================================================================
==658729==ERROR: AddressSanitizer: heap-use-after-free on address 0x608000047d20 at pc 0x561db6768e84 bp 0x7fff3f1289b0 sp 0x7fff3f1289a0
READ of size 8 at 0x608000047d20 thread T0
    #0 0x561db6768e83 in CVInfoDrawText ../fontforgeexe/charview.c:4110
    #1 0x561db6781833 in InfoExpose ../fontforgeexe/charview.c:5952
    #2 0x561db6786e4e in cv_e_h ../fontforgeexe/charview.c:6289
    #3 0x561db6daf57b in _GWidget_Container_eh ../gdraw/gcontainer.c:281
    #4 0x561db6db54b9 in _GWidget_TopLevel_eh ../gdraw/gcontainer.c:743
    #5 0x561db6de7fa1 in _GGDKDraw_CallEHChecked ../gdraw/ggdkdraw.c:346
    #6 0x561db6df0ff9 in _GGDKDraw_DispatchEvent ../gdraw/ggdkdraw.c:1249
    #7 0x7fba221ac233  (/usr/lib/libgdk-3.so.0+0x95233)
    #8 0x7fba2219ea51  (/usr/lib/libgdk-3.so.0+0x87a51)
    #9 0x7fba2219e43e  (/usr/lib/libgdk-3.so.0+0x8743e)
    #10 0x7fba2219e60c  (/usr/lib/libgdk-3.so.0+0x8760c)
    #11 0x7fba21e12809 in g_signal_emit_valist (/usr/lib/libgobject-2.0.so.0+0x2f809)
    #12 0x7fba21e1297f in g_signal_emit (/usr/lib/libgobject-2.0.so.0+0x2f97f)
    #13 0x7fba221a839f  (/usr/lib/libgdk-3.so.0+0x9139f)
    #14 0x7fba221b6ceb  (/usr/lib/libgdk-3.so.0+0x9fceb)
    #15 0x7fba21d0a763  (/usr/lib/libglib-2.0.so.0+0x52763)
    #16 0x7fba21d0a33f in g_main_context_dispatch (/usr/lib/libglib-2.0.so.0+0x5233f)
    #17 0x7fba21d57fa8  (/usr/lib/libglib-2.0.so.0+0x9ffa8)
    #18 0x7fba21d09220 in g_main_context_iteration (/usr/lib/libglib-2.0.so.0+0x51220)
    #19 0x561db6df822f in GGDKDrawEventLoop ../gdraw/ggdkdraw.c:2205
    #20 0x561db6dc1547 in GDrawEventLoop ../gdraw/gdraw.c:731
    #21 0x561db6d1fa16 in fontforge_main ../fontforgeexe/startui.c:1413
    #22 0x561db6682d58 in main ../fontforgeexe/main.c:33
    #23 0x7fba21b18001 in __libc_start_main (/usr/lib/libc.so.6+0x27001)
    #24 0x561db6682c6d in _start (/usr/bin/fontforge+0x159c6d)

0x608000047d20 is located 0 bytes inside of 96-byte region [0x608000047d20,0x608000047d80)
freed by thread T0 here:
    #0 0x7fba241520e9 in __interceptor_free /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cpp:123
    #1 0x7fba237772a2 in SplinePointFree ../fontforge/splineutil.c:189
    #2 0x7fba237775c0 in SplinePointsFree ../fontforge/splineutil.c:227
    #3 0x7fba23777777 in SplinePointListFree ../fontforge/splineutil.c:248
    #4 0x7fba236f054f in MergeOpenAndFreeClosed ../fontforge/splineoverlap.c:3384
    #5 0x7fba236f5604 in SplineSetRemoveOverlap ../fontforge/splineoverlap.c:3999
    #6 0x561db67aa9bd in _CVMenuOverlap ../fontforgeexe/charview.c:9810
    #7 0x561db67aacd4 in CVMenuOverlap ../fontforgeexe/charview.c:9819
    #8 0x561db6e438ad in gmenu_mouse ../gdraw/gmenu.c:948
    #9 0x561db6e47661 in gmenu_eh ../gdraw/gmenu.c:1389
    #10 0x561db6de7fa1 in _GGDKDraw_CallEHChecked ../gdraw/ggdkdraw.c:346
    #11 0x561db6df0ff9 in _GGDKDraw_DispatchEvent ../gdraw/ggdkdraw.c:1249
    #12 0x7fba221ac233  (/usr/lib/libgdk-3.so.0+0x95233)

previously allocated by thread T0 here:
    #0 0x7fba24152639 in __interceptor_calloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cpp:154
    #1 0x7fba23776ebd in SplinePointCreate ../fontforge/splineutil.c:166
    #2 0x7fba2386bae0 in SVGParsePath ../fontforge/svg.c:1389
    #3 0x7fba2386e34b in SVGParseExtendedPath ../fontforge/svg.c:1583
    #4 0x7fba2387a934 in _SVGParseSVG ../fontforge/svg.c:2761
    #5 0x7fba2387a4ac in _SVGParseSVG ../fontforge/svg.c:2722
    #6 0x7fba2387b7c1 in SVGParseSVG ../fontforge/svg.c:2845
    #7 0x7fba23884eea in EntityInterpretSVG ../fontforge/svg.c:3708
    #8 0x7fba2388513b in SplinePointListInterpretSVG ../fontforge/svg.c:3726
    #9 0x7fba230d22c3 in SCImportSVG ../fontforge/cvimages.c:359
    #10 0x561db68885ff in ImportSVG ../fontforgeexe/cvimportdlg.c:77
    #11 0x561db6890bc0 in GFD_ImportOk ../fontforgeexe/cvimportdlg.c:632
    #12 0x561db6d91db8 in GButtonInvoked ../gdraw/gbuttons.c:257
    #13 0x561db6d95208 in gbutton_mouse ../gdraw/gbuttons.c:485
    #14 0x561db6dafc17 in _GWidget_Container_eh ../gdraw/gcontainer.c:304
    #15 0x561db6db54b9 in _GWidget_TopLevel_eh ../gdraw/gcontainer.c:743
    #16 0x561db6de7fa1 in _GGDKDraw_CallEHChecked ../gdraw/ggdkdraw.c:346
    #17 0x561db6df0ff9 in _GGDKDraw_DispatchEvent ../gdraw/ggdkdraw.c:1249
    #18 0x7fba221ac233  (/usr/lib/libgdk-3.so.0+0x95233)

SUMMARY: AddressSanitizer: heap-use-after-free ../fontforgeexe/charview.c:4110 in CVInfoDrawText
```
This suggests an as-yet-unidentified problem in `MergeOpenAndFreeClosed`. Interestingly the crash doesn't appear to happen when no points are selected.

### PR DESCRIPTION
**Pull request by frank-trampe**
Closes #4380.

### CODE REVIEW

**Review by skef — APPROVED** [review]
lgtm

**Comment by frank-trampe** [comment]
@skef, does it pass your tests from #4380?

File: fontforgeexe/charview.c [hunk 1]
```diff
@@ -9812,14 +9812,14 @@ static void _CVMenuOverlap(CharView *cv,enum overlap_type ot) {
     if ( cv->b.sc->inspiro && hasspiro()) {
 	// Detection is not implemented for Spiro, so just clear them.
 	// TODO: Detect point survival in Spiro mode.
-	cv->lastselpt = NULL; cv->lastselcp = NULL;
+	cv->p.sp = cv->lastselpt = NULL; cv->p.spiro = cv->lastselcp = NULL;
```
Comment (skef) [inline fontforgeexe/charview.c:9815]: This is weird stylistically, seems like one chained assignment or three separate assignments would be preferable.

**Comment by skef** [comment]
On the narrow question, yes. There is now no ASAN problem with these changes. (Some overlap may not be removed but that's not what #4380 was about.)

File: fontforgeexe/charview.c (hunk 1 above)
Comment (frank-trampe) [reply fontforgeexe/charview.c]: I added a linebreak to resolve this in a way compatible with the parallel code below.

**Comment by frank-trampe** [comment]
Even if overlap removal is perfect, it may remove some points, which in may in turn be selected, so we need to fix this either way.

**Comment by skef** [comment]
@frank-trampe yes -- I think the only question is whether to open another RemoveOverlap completeness issue for the SVG in #4380 but I'm happy not to.

**Comment by frank-trampe** [comment]
I'll work on that and a few other remove overlap issues tonight and tomorrow. Much more pleasant than dealing with CID issues.

**Comment by frank-trampe** [comment]
@skef, I just tried removing the overlap from the sample in #4380, and the result appears to be correct.

**Comment by skef** [comment]
are adjacent lines not supposed to be removed? I was wondering about the bottom points.

**Comment by frank-trampe** [comment]
It looked fine to me, but I might be missing something. Post an annotated screenshot, perhaps?

**Comment by skef** [comment]
overlap

This is the bottom point of the heart.

**Comment by frank-trampe** [comment]
That's odd. It worked perfectly for me. Post your SFD so that we can rule out SVG import differences. I have posted mine.

[broken_heart_preoperative_1_1.sfd.gz](https://github.com/fontforge/fontforge/files/5504286/broken_heart_preoperative_1_1.sfd.gz)
[broken_heart_postoperative_1_1.sfd.gz](https://github.com/fontforge/fontforge/files/5504285/broken_heart_postoperative_1_1.sfd.gz)

**Comment by skef** [comment]
Sorry, try importing without scale to fit.

**Comment by frank-trampe** [comment]
That does indeed allow replicating the problem. I have a sneaking feeling that a scale-sensitive problem is going to be particularly difficult to fix.

**Comment by skef** [comment]
Yeah, there are several of those in the expand stroke code. I wouldn't worry about it too much unless you feel motivated to.

**Pull request merged**